staging/src/k8s.io/apiserver/pkg/audit/policy/checker.go

```diff
@@ -163,6 +161,11 @@ func ruleMatchesResource(r *audit.PolicyRule, attrs authorizer.Attributes) bool
 						return true
 					}
 				}
+				if strings.HasSuffix(res, "/*") {
+					if strings.HasPrefix(resource, strings.TrimRight(res, "/*")) {
```

Comment:
Rather than check the prefix, I think this is more safe:
attrs.GetResource() == strings.TrimRight(res, "/*")

Reply:
I think they are the same 
```
	resource := attrs.GetResource()
	// If subresource, the resource in the policy must match "(resource)/(subresource)"
	if sr := attrs.GetSubresource(); sr != "" {
		resource = resource + "/" + sr
	}
```

Reply:
They are not 100% same.
What if user has such illegal resource in policy: "pods/log/*" ?

Reply:
Should we support this illegal resource? I think we should not support wrong config.

Reply:
> Should we support this illegal resource? I think we should not support wrong config.

With ` attrs.GetResource() == strings.TrimRight(res, "/*")` we can avoid this, so why not?

Reply:
We validate policy format here:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/apis/audit/validation/validation.go

Reply:
I know your meaning .